### PR TITLE
Add Type.Data.Ordering; Reexport Data.Symbols, Type.Equality

### PR DESCRIPTION
**Pull request by LiamGoodacre**

### CODE REVIEW

File: src/Type/Prelude.purs [hunk 1]
```diff
@@ -1,5 +1,11 @@
 module Type.Prelude
-  ( module Proxy
+  ( module Equality
+  , module Data.Ordering
+  , module Proxy
+  , module Data.Symbol
```
Comment (garyb) [inline src/Type/Prelude.purs]: Minor thing, but we don't need to alias modules for export, if you just have

```
import Type.Equality (class TypeEquals, from, to)
```

then exporting `Type.Equality` works, and only exports the imported members.

**Comment by LiamGoodacre** [comment]
(Note: this obviously this won't compile yet: https://github.com/purescript/purescript/pull/2486)

File: src/Type/Data/Ordering.purs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+class IsOrdering (ordering :: Ordering) where
+  reflectOrdering :: OProxy ordering -> Ordering
+
+instance isOrderingLT :: IsOrdering LT where reflectOrdering _ = LT
+instance isOrderingEQ :: IsOrdering EQ where reflectOrdering _ = EQ
+instance isOrderingGT :: IsOrdering GT where reflectOrdering _ = GT
+
+-- | Use a value level `Ordering` as a type-level `Ordering`
+reifyOrdering :: forall r. Ordering -> (forall o. IsOrdering o => OProxy o -> r) -> r
+reifyOrdering o f = coerce f { reflectOrdering: \_ -> o } OProxy where
```
Comment (paf31) [inline src/Type/Data/Ordering.purs]: You can just use pattern matching here, no need for the unsafe coerce.

Comment (LiamGoodacre) [inline src/Type/Data/Ordering.purs]: Updated :) - I totally just copied `reifySymbol`

**Comment by paf31** [comment]
I'll merge this and add a comment on the README, explaining the status of the library.

**Pull request merged**